LandlordAPI/src/test/java/com/qa/landlordGateway/testMain/controllerTest/LandlordControllerTest.java

```diff
@@ -0,0 +1,97 @@
+package com.qa.landlordGateway.testMain.controllerTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.mockito.ArgumentMatchers.notNull;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureWebMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.qa.landlordGateway.controllers.LandlordController;
+import com.qa.landlordGateway.entity.Landlord;
+import com.qa.landlordGateway.entity.LandlordBuilder;
+import com.qa.landlordGateway.service.LandlordService;
+import com.qa.landlordGateway.testMain.Constants;
+
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(LandlordController.class)
+@AutoConfigureWebMvc
+public class LandlordControllerTest {
+
+	@Autowired
+	private MockMvc mockMvc;
```

Comment:
Necessary field injection?

Reply:
It was not, Removed it's now a constructor injection